### PR TITLE
(installer): pin default versions of APM libraries

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
In the effort to avoid breaking changes for customers we are proposing a default pin of the library versions. In addition to this change we will be updating the default install command in the UI and public documentation to include any new versions of libraries.

By pinning to the major tag: libraries can still release bug fixes and add new features to customers; and customer will not receive breaking changes (like dropped support of runtimes/integrations).

[](https://datadoghq.atlassian.net/browse/APMON-1069)

### CODE REVIEW

File: pkg/fleet/installer/default_packages.go [hunk 1]
```diff
@@ -41,6 +41,15 @@ var packageDependencies = map[string][]string{
 	"datadog-apm-library-python": {"datadog-apm-inject"},
 }
 
+var packageDefaultVersions = map[string]string{
+	"datadog-apm-inject":         "latest",
+	"datadog-apm-library-java":   "1-1",
+	"datadog-apm-library-ruby":   "1-1",
+	"datadog-apm-library-js":     "5-1",
+	"datadog-apm-library-dotnet": "2-1",
+	"datadog-apm-library-python": "2-1",
```
[the remark this answers is not in the document]
Comment (Kyle-Verhoog) [reply pkg/fleet/installer/default_packages.go:50]: Got it now:

```
~ ❯❯❯ crane ls gcr.io/datadoghq/apm-library-js-package
...
5
5.15.0-1
5.16
5.16.0-1
5.17
5.17.0-1
latest
```

[the remark this answers is not in the document]
Comment (raphaelgavache) [reply pkg/fleet/installer/default_packages.go:50]: you need to remove the `-1` at the end

**Comment by raphaelgavache** [comment]
can you add a test to validate in this test suite it works: https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/tests/installer/package_apm_inject_test.go#L25-L27

File: pkg/fleet/installer/default_packages.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (raphaelgavache) [reply pkg/fleet/installer/default_packages.go:50]: or does it work with `-1`, the tags from the container registries don't have it on major/minor versions

[the remark this answers is not in the document]
Comment (Kyle-Verhoog) [reply pkg/fleet/installer/default_packages.go:50]: oh weird, the major tags don't have the `-1` 🤔 idk when it's added and when it's not lol. I will verify with the e2e tests.

[the remark this answers is not in the document]
Comment (raphaelgavache) [reply pkg/fleet/installer/default_packages.go:50]: the code creating it /publishing minor/major/latest/default version and setting`-1` is here: https://github.com/DataDog/agent-release-management/pull/197
I'm confused in general about it, I'll just keep using latest and majors

**Comment by paullegranddc** [comment]
Closing an reopening because of the rebase

**Pull request closed without merging**